crates/orderbook/src/orderbook.rs

```diff
@@ -243,13 +274,61 @@ fn set_available_balances(orders: &mut [Order], cache: &SolvableOrdersCache) {
     }
 }
 
+async fn get_orders_with_native_prices(
+    mut orders: Vec<Order>,
+    native_price_estimator: &dyn NativePriceEstimating,
+) -> (Vec<Order>, BTreeMap<H160, U256>) {
+    let traded_tokens = orders
+        .iter()
+        .flat_map(|order| {
+            [
+                order.order_creation.sell_token,
+                order.order_creation.buy_token,
+            ]
+        })
+        .collect::<HashSet<_>>()
+        .into_iter()
+        .collect::<Vec<_>>();
+    let prices = native_price_estimator
+        .estimate_native_prices(&traded_tokens)
+        .await
+        .into_iter()
+        .zip(traded_tokens)
+        .filter_map(|(price, token)| match price {
+            Ok(price) => Some((token, to_normalized_price(price))),
+            Err(err) => {
+                tracing::warn!(?token, ?err, "error estimating native token price");
+                None
+            }
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    orders.retain(|order| {
+        prices.contains_key(&order.order_creation.sell_token)
+            && prices.contains_key(&order.order_creation.buy_token)
+    });
+
+    (orders, prices)
+}
+
+fn to_normalized_price(price: f64) -> U256 {
+    // NOTE: The `NativePriceEstimating` component returns prices denominated
+    // in the token that it is estimating and not in ETH. This means that we
+    // need to invert the price in order for it to be correct.
+    U256::from_f64_lossy(1e18 / price)
+}
```

Comment:
This can panic if the price is 0 which is allowed by the price estimator interface. I think we should filter 0 prices out.
As a second point, maybe it makes sense to also guarantee that prices in the auction endpoint itself are never zero so we would have to check here that the U256 didn't round to 0.

Reply:
Changed to check that the resulting `U256` prices are valid.